homeassistant/components/philips_js/device_trigger.py

```diff
@@ -0,0 +1,60 @@
+"""Provides device automations for control of device."""
+from typing import List
+
+import voluptuous as vol
+
+from homeassistant.components.automation import AutomationActionType
+from homeassistant.components.device_automation import TRIGGER_BASE_SCHEMA
+from homeassistant.components.homeassistant.triggers import event as event_trigger
+from homeassistant.const import (
+    ATTR_DEVICE_ID,
+    CONF_DEVICE_ID,
+    CONF_DOMAIN,
+    CONF_PLATFORM,
+    CONF_TYPE,
+)
+from homeassistant.core import CALLBACK_TYPE, HomeAssistant
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DOMAIN, EVENT_TURN_ON
+
+TRIGGER_TYPES = {"turn_on"}
+TRIGGER_SCHEMA = TRIGGER_BASE_SCHEMA.extend(
+    {
+        vol.Required(CONF_TYPE): vol.In(TRIGGER_TYPES),
+    }
+)
+
+
+async def async_get_triggers(hass: HomeAssistant, device_id: str) -> List[dict]:
+    """List device triggers for device."""
+    triggers = []
+    triggers.append(
+        {
+            CONF_PLATFORM: "device",
+            CONF_DEVICE_ID: device_id,
+            CONF_DOMAIN: DOMAIN,
+            CONF_TYPE: "turn_on",
```

Comment:
Several of these are available for import from homeassistant.const rather than needing to use the raw strings here (same applies to a few other raw strings in this PR).

Reply:
They are not really the same thing as the global constant. We shouldn't use a global constant for a different meaning, even thou the value of the constant is the same.

Reply:
Why is it not the same thing?

Reply:
I fixed up turn on to have a constant. "device" is like that every where. There don't seem to be a constant for the string "device" with the meaning of a platform name.

For reference: 

```
CONST_DEVICE_FRIENDLY_NAME = "device"
CONST_DEVICE_ID = "device"
```

Event thou these two constant declare to the same string. They shouldn't be used arbitrarily. One signifies some name that might be user visible, one signifies a identifier. It's important that the constant name declares what it signifies even thou they share the same underlying data.